iroh-net/src/bin/derper.rs

```diff
@@ -206,23 +151,100 @@ fn load_private_key(filename: impl AsRef<Path>) -> Result<rustls::PrivateKey> {
+    /// This field is only read in we are serving the derper over HTTPS. In that case, we must listen for requests for the `/generate_204` over a non-TLS connection.
+    captive_portal_port: u16,
+    /// The UDP port on which to serve STUN. The listener is bound to the same IP (if any) as specified in the `addr` field. Defaults to 3478.
+    stun_port: u16,
+    /// Mode for getting a cert. possible options: 'Manual', 'LetsEncrypt'
+    /// When using manual mode, a certificate will be read from `<hostname>.crt` and a private key from
+    /// `<hostname>.key`, with the `<hostname>` being the escaped hostname.
+    cert_mode: CertMode,
+    /// Whether to use the LetsEncrypt production or staging server.
+    ///
+    /// While in developement, LetsEncrypt prefers you to use the staging server. However, the staging server seems to
+    /// only use `ECDSA` keys. In their current set up, you can only get intermediate certificates
+    /// for `ECDSA` keys if you are on their "allowlist". The production server uses `RSA` keys,
+    /// which allow for issuing intermediate certificates in all normal circumstances.
+    /// So, to have valid certificates, we must use the LetsEncrypt production server.
+    /// Read more here: <https://letsencrypt.org/certificates/#intermediate-certificates>
+    /// Default is true. This flag is ignored if we are not using `cert_mode: CertMode::LetsEncrypt`.
+    prod_tls: bool,
+    /// The contact email for the tls certificate.
+    ///
+    /// Default is `None`, must be set when running the Derper with TLS
+    /// enabled.
+    tls_contact: Option<String>,
+    /// Directory to store LetsEncrypt certs or read certificates from, if TLS is used.
+    cert_dir: Option<PathBuf>,
+    /// Certificate hostname. Defaults to `derp.iroh.network`
+    hostname: String,
+    /// Whether to run a STUN server. It will bind to the same IP as the `addr` field.
+    ///
+    /// Defaults to `true`.
+    run_stun: bool,
+    /// Whether to run a DERP server. The only reason to set this false is if you're decommissioning a
+    /// server but want to keep its bootstrap DNS functionality still running.
+    ///
+    /// Defaults to `true`
+    run_derp: bool,
+    /// If non-empty, path to file containing the mesh pre-shared key file. It should contain some hex string; whitespace is trimmed.
+    mesh_psk_file: Option<PathBuf>,
+    /// Comma-separated list of urls to mesh with. Must also include the scheme ('http' or
+    /// 'https').
+    ///
+    /// If there is no `mesh_psk_file` with a valid `MeshKey`, this field is ignored.
+    mesh_with: Vec<Url>,
+    /// Optional comma-separated list of hostnames to make available at /bootstrap-dns.
+    bootstrap_dns_names: Option<Vec<Url>>,
+    /// Optional comma-separated list of hostnames to make available at /bootstrap-dns and not publish in the list
+    unpublished_dns_names: Option<Vec<Url>>,
```

Comment:
I believe this and the above option are unused and can be removed

Reply:
into the trash they go